compilers/concrete-optimizer/concrete-optimizer/src/optimization/dag/multi_parameters/analyze.rs

```diff
@@ -163,10 +210,15 @@ pub fn original_instrs_partition(
 fn out_variance(
     op: &unparametrized::UnparameterizedOperator,
     out_shapes: &[Shape],
-    out_variances: &Vec<Vec<SymbolicVariance>>,
+    out_variances: &[Vec<SymbolicVariance>],
     nb_partitions: usize,
     instr_partition: &InstructionPartition,
+    input_override: Option<Vec<SymbolicVariance>>,
```

Comment:
why copying the Option (and Vec) here ? can't you use a ref and only clone the symbolic variance if needed ?

Reply:
Vec is a copy, Option is a copy

Reply:
Well, in rust sense, there is no copy here, it is a move of an existing value. In the calling code, the `input_override` is just created for this function to call and not used after, so no copy occurs. I mean, in non-optimized code, maybe you have an on-stack copy of a pointer (which the `Vec` is) but nothing like allocating a new heap storage and copying the elements.